src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -12624,7 +12625,10 @@ void VRFTerminalUnitEquipment::ControlVRF_FluidTCtrl(EnergyPlusData &state,
         if (NoCompOutput <= QZnReq) return;
     } else if (VRFCoolingMode || HRCoolingMode) {
         // IF the system is in cooling mode and/or the terminal unit requests cooling
-        if (NoCompOutput <= QZnReq) return;
+        if (NoCompOutput <= QZnReq || (QZnReq >= -1.0*DataHVACGlobals::SmallLoad && !HRCoolingMode)) {
+            state.dataHVACVarRefFlow->VRFTU(VRFTUNum).coolingCoilActive = false;
+            if (!this->SuppHeatingCoilPresent) return;
+        }
```

Comment:
It seems to me this logic is the issue and you can get rid of the other changes here. This logic tests that the coil off output (NoCompOutput) is less than the load (QZnReq) and returns. But if there is a zone heating load then QZnReq > 0 and this logic fails (because NoCompOutput <= QZnReq). You need to fall through here for heating loads. Then below were PLR = 1 is set, you also need to avoid turning on cooling. At least that's what I think should work here.